client/lib/paths/login/index.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Internal dependencies
+ */
+import { addQueryArgs } from 'lib/url';
+import config, { isEnabled } from 'config';
+
+export function login( { redirectTo } ) {
+	const url = isEnabled( 'wp-login' ) ? '/login' : config( 'login_url' );
+
+	return redirectTo
+		? addQueryArgs( { redirect_to: redirectTo }, url )
```

Comment:
I'll add tests to double check, but it uses `format` method from `node-url` library, so it should cover this use case.